### PR TITLE
Add 2.10.18 Release Notes

### PR DESCRIPTION
**Pull request by billpalombi**
I've removed a bunch of trivial docs PRs

Corrects one header in a previous release.

@WillRaphaelson I've adding the highlight section for you to fill out.

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,45 @@
...
+
+### Fixes
+- Fix pull step saving by preserving placeholders with missing values — https://github.com/PrefectHQ/prefect/pull/10053
+- Fix windows file limit check — https://github.com/PrefectHQ/prefect/pull/10059
+- Allows use of Prefect variables in `job_variables` section of deploy config in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10078
+- Add default option to new_parameters.pop in `explode_variadic_parameter` used to handle `**kwargs` in task mapping — https://github.com/PrefectHQ/prefect/pull/10067
+- Update Docker-based `prefect init` recipes to use `push_docker_image` step — https://github.com/PrefectHQ/prefect/pull/10092
+- Update `prefect deploy` to skip schedule prompts if `schedule` key present in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10074
+- Fix saving of `pull` and `push` step deployment configuration — https://github.com/PrefectHQ/prefect/pull/10087
+
```
Comment (znicholasbrown) [inline RELEASE-NOTES.md:21]:
```suggestion
- Fix issue hosting and running the UI in unsecured contexts - https://github.com/PrefectHQ/prefect-design/pull/829

```

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,59 @@
 # Prefect Release Notes
 
+## Release 2.10.18
+
+### Event driven deployments with triggers
+Deployments in Prefect Cloud now have an optional triggers field, which describes what events (or lack thereof) should trigger a flow run. You can add triggers to deployments in your .yaml files or in the UI from your deployment page. Deployment triggers use automations under the hood, and any automations that trigger a deployment will be linked to from the deployments page. 
+
```
Comment (serinamarie) [inline RELEASE-NOTES.md:7]:
```suggestion
See the following pull request for implementation details:
- https://github.com/PrefectHQ/prefect/pull/10049
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,59 @@
...
+Not only that, we will also create a `pull` step for you when you choose to build a Docker image through `prefect deploy`. Whether you have your own Dockerfile or you want to use the auto-build feature in `build_docker_image`, we will create a `pull` step for you to help you set the correct path to your flow code.
+
+See the following pull request for implementation details:
+- https://github.com/PrefectHQ/prefect/pull/10022
+- https://github.com/PrefectHQ/prefect/pull/10090
+
+### Enhancements
+- Allow saving of updated deployment configurations — https://github.com/PrefectHQ/prefect/pull/10018
+- Add support for deployment triggers to `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10049
```
Comment (serinamarie) [inline RELEASE-NOTES.md:23]: Moved to highlight
```suggestion
```

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,59 @@
...
+- Add `prefect.yaml` `pull` step generation based on `build_docker_image` `build` step  — https://github.com/PrefectHQ/prefect/pull/10090
```
Comment (serinamarie) [inline RELEASE-NOTES.md:24]: Referenced in highlight
```suggestion
```

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,59 @@
...
+- Webhook creation UI (cloud only)
```
Comment (serinamarie) [inline RELEASE-NOTES.md:25]: ?

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,5 +1,59 @@
...
+
+### Fixes
+- Fix pull step saving by preserving placeholders with missing values — https://github.com/PrefectHQ/prefect/pull/10053
```
Comment (serinamarie) [inline RELEASE-NOTES.md:28]:
```suggestion
- Fix deployment `pull` step saving by preserving placeholders with missing values — https://github.com/PrefectHQ/prefect/pull/10053
```

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,5 +1,59 @@
...
+- Fix windows file limit check — https://github.com/PrefectHQ/prefect/pull/10059
+- Allows use of Prefect variables in `job_variables` section of deploy config in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10078
+- Add default option to new_parameters.pop in `explode_variadic_parameter` used to handle `**kwargs` in task mapping — https://github.com/PrefectHQ/prefect/pull/10067
+- Update Docker-based `prefect init` recipes to use `push_docker_image` step — https://github.com/PrefectHQ/prefect/pull/10092
```
Comment (serinamarie) [inline RELEASE-NOTES.md:32]: This is also not a fix

File: RELEASE-NOTES.md [hunk 8]
```diff
@@ -1,5 +1,59 @@
...
+- Allow saving of updated deployment configurations — https://github.com/PrefectHQ/prefect/pull/10018
+- Add support for deployment triggers to `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10049
+- Add `prefect.yaml` `pull` step generation based on `build_docker_image` `build` step  — https://github.com/PrefectHQ/prefect/pull/10090
+- Webhook creation UI (cloud only)
+
+### Fixes
+- Fix pull step saving by preserving placeholders with missing values — https://github.com/PrefectHQ/prefect/pull/10053
+- Fix windows file limit check — https://github.com/PrefectHQ/prefect/pull/10059
+- Allows use of Prefect variables in `job_variables` section of deploy config in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10078
+- Add default option to new_parameters.pop in `explode_variadic_parameter` used to handle `**kwargs` in task mapping — https://github.com/PrefectHQ/prefect/pull/10067
```
Comment (serinamarie) [inline RELEASE-NOTES.md:31]:
```suggestion
- Add default option to `new_parameters.pop` in `explode_variadic_parameter` used to handle `**kwargs` in task mapping — https://github.com/PrefectHQ/prefect/pull/10067
```

File: RELEASE-NOTES.md [hunk 9]
```diff
@@ -1,5 +1,59 @@
...
+- Update Docker-based `prefect init` recipes to use `push_docker_image` step — https://github.com/PrefectHQ/prefect/pull/10092
+- Update `prefect deploy` to skip schedule prompts if `schedule` key present in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10074
+- Fix saving of `pull` and `push` step deployment configuration — https://github.com/PrefectHQ/prefect/pull/10087
+
+### Documentation
+- Adjust docs to reflect Prefect requires Python 3.8 — https://github.com/PrefectHQ/prefect/pull/9853
+- Add custom pull step examples with variables — https://github.com/PrefectHQ/prefect/pull/10073
```
Comment (serinamarie) [inline RELEASE-NOTES.md:39]:
```suggestion
- Add custom `pull` step examples to deployment management docs — https://github.com/PrefectHQ/prefect/pull/10073
```

File: RELEASE-NOTES.md [hunk 10]
```diff
@@ -1,5 +1,60 @@
...
+- Add example for finding Prefect Cloud Account ID and Workspace ID — https://github.com/PrefectHQ/prefect/pull/10103
+- Add links to Webhooks documentation from Events documentation — https://github.com/PrefectHQ/prefect/pull/10045
+- Simply deployment description — https://github.com/PrefectHQ/prefect/pull/10050
+- Add trigger to deployment fields — https://github.com/PrefectHQ/prefect/pull/10097
+
+### Contributors
+- @masonmenges made their first contribution in https://github.com/PrefectHQ/prefect/pull/9977
+- @garylavayou made their first contribution in https://github.com/PrefectHQ/prefect/pull/10060
+- @themattmorris made their first contribution in https://github.com/PrefectHQ/prefect/pull/10056
+- @aimeemcmanus made their first contribution in https://github.com/PrefectHQ/prefect/pull/10076
```
Comment (serinamarie) [inline RELEASE-NOTES.md:50]:
```suggestion
- @garylavayou made their first contribution in https://github.com/PrefectHQ/prefect/pull/10060
- @themattmorris made their first contribution in https://github.com/PrefectHQ/prefect/pull/10056
```
excluding Prefect employees, despite their contributions being very much appreciated

File: RELEASE-NOTES.md [hunk 11]
```diff
@@ -1,5 +1,60 @@
...
+- @NodeJSmith
+- @garylavayou
+- @rpeden
+- @themattmorris
```
Comment (serinamarie) [inline RELEASE-NOTES.md:54]:
```suggestion
- @NodeJSmith
- @rpeden
```
aforementioned

File: RELEASE-NOTES.md [hunk 12]
```diff
@@ -1,5 +1,60 @@
...
+- Fix saving of `pull` and `push` step deployment configuration — https://github.com/PrefectHQ/prefect/pull/10087
+
+### Documentation
+- Adjust docs to reflect Prefect requires Python 3.8 — https://github.com/PrefectHQ/prefect/pull/9853
+- Add custom pull step examples with variables — https://github.com/PrefectHQ/prefect/pull/10073
+- Add troubleshooting guide to docs — https://github.com/PrefectHQ/prefect/pull/10079
+- Add example for finding Prefect Cloud Account ID and Workspace ID — https://github.com/PrefectHQ/prefect/pull/10103
+- Add links to Webhooks documentation from Events documentation — https://github.com/PrefectHQ/prefect/pull/10045
+- Simply deployment description — https://github.com/PrefectHQ/prefect/pull/10050
+- Add trigger to deployment fields — https://github.com/PrefectHQ/prefect/pull/10097
```
Comment (serinamarie) [inline RELEASE-NOTES.md:44]: Related to deployment trigger highlight, should be moved up there

Comment (serinamarie) [inline RELEASE-NOTES.md:44]:
```suggestion
```

Related to deployment trigger highlight, should go up there

File: RELEASE-NOTES.md [hunk 13]
```diff
@@ -1,5 +1,60 @@
...
+- Update `prefect deploy` to skip schedule prompts if `schedule` key present in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10074
+- Fix saving of `pull` and `push` step deployment configuration — https://github.com/PrefectHQ/prefect/pull/10087
+
+### Documentation
+- Adjust docs to reflect Prefect requires Python 3.8 — https://github.com/PrefectHQ/prefect/pull/9853
+- Add custom pull step examples with variables — https://github.com/PrefectHQ/prefect/pull/10073
+- Add troubleshooting guide to docs — https://github.com/PrefectHQ/prefect/pull/10079
+- Add example for finding Prefect Cloud Account ID and Workspace ID — https://github.com/PrefectHQ/prefect/pull/10103
+- Add links to Webhooks documentation from Events documentation — https://github.com/PrefectHQ/prefect/pull/10045
+- Simply deployment description — https://github.com/PrefectHQ/prefect/pull/10050
```
Comment (serinamarie) [inline RELEASE-NOTES.md:43]:
```suggestion
- Simplify deployment description in docs — https://github.com/PrefectHQ/prefect/pull/10050
```

File: RELEASE-NOTES.md [hunk 14]
```diff
@@ -1,5 +1,60 @@
...
+- Add default option to new_parameters.pop in `explode_variadic_parameter` used to handle `**kwargs` in task mapping — https://github.com/PrefectHQ/prefect/pull/10067
+- Update Docker-based `prefect init` recipes to use `push_docker_image` step — https://github.com/PrefectHQ/prefect/pull/10092
+- Update `prefect deploy` to skip schedule prompts if `schedule` key present in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10074
+- Fix saving of `pull` and `push` step deployment configuration — https://github.com/PrefectHQ/prefect/pull/10087
+
+### Documentation
+- Adjust docs to reflect Prefect requires Python 3.8 — https://github.com/PrefectHQ/prefect/pull/9853
+- Add custom pull step examples with variables — https://github.com/PrefectHQ/prefect/pull/10073
+- Add troubleshooting guide to docs — https://github.com/PrefectHQ/prefect/pull/10079
+- Add example for finding Prefect Cloud Account ID and Workspace ID — https://github.com/PrefectHQ/prefect/pull/10103
```
[the remark this answers is not in the document]
Comment (serinamarie) [reply RELEASE-NOTES.md:41]: it wasn't a true example, so I changed the wording

File: RELEASE-NOTES.md [hunk 15]
```diff
@@ -1,5 +1,60 @@
 # Prefect Release Notes
 
+## Release 2.10.18
+
+### Event driven deployments with triggers
```
Comment (serinamarie) [inline RELEASE-NOTES.md:5]:
```suggestion
### Event-driven deployments with triggers
```
I like the dash :-)

File: RELEASE-NOTES.md [hunk 16]
```diff
@@ -1,5 +1,60 @@
...
+Deployments in Prefect Cloud now have an optional triggers field, which describes what events (or lack thereof) should trigger a flow run. You can add triggers to deployments in your .yaml files or in the UI from your deployment page. Deployment triggers use automations under the hood, and any automations that trigger a deployment will be linked to from the deployments page.
```
[the remark this answers is not in the document]
Comment (serinamarie) [reply RELEASE-NOTES.md:6]: I actually don't feel we should begin with a Prefect Cloud highlight. I think we should move this down.

File: RELEASE-NOTES.md [hunk 17]
```diff
@@ -1,5 +1,56 @@
...
+### Docker image support during flow deployment
+We have introduced enhanced support for Docker-based infrastructures when deploying flows through the interactive `prefect deploy` experience. Users can now easily custom-build or auto-build Docker images and push them to remote registries if they so choose.
+
+The CLI automatically detects if a work pool supports Docker images (e.g., docker, ecs, cloud-run) during `prefect deploy`, guiding the user through the experience of building and pushing a Docker image.
+
+This enhancement to managing deployments will greatly simplify the process of creating the necessary `build` and `push` steps for deployments.
+
+Not only that, we will also create a `pull` step for you when you choose to build a Docker image through `prefect deploy`. Whether you have your own Dockerfile or you want to use the auto-build feature in `build_docker_image`, we will create a `pull` step for you to help you set the correct path to your flow code.
+
+See the following pull request for implementation details:
```
Comment (serinamarie) [inline RELEASE-NOTES.md:14]:
```suggestion
See the following pull requests for implementation details:
```

File: RELEASE-NOTES.md [hunk 18]
```diff
@@ -1,5 +1,56 @@
...
+- https://github.com/PrefectHQ/prefect/pull/10022
+- https://github.com/PrefectHQ/prefect/pull/10090
+
+### Event-driven deployments with triggers
+You can now add event-based triggers to your deployments in Prefect Cloud, allowing you to add triggers into your `prefect.yaml` file or from the Prefect UI deployment page. Deployment triggers use automations under the hood, and any automations that trigger a deployment will be linked to from the deployments page. 
+See the following pull request for implementation details:
```
Comment (serinamarie) [inline RELEASE-NOTES.md:20]:
```suggestion

See the following pull requests for implementation details:
```

**Review by serinamarie — APPROVED** [review]
Left a few more comments, but after that LGTM

File: RELEASE-NOTES.md [hunk 19]
```diff
@@ -1,5 +1,56 @@
 # Prefect Release Notes
 
+## Release 2.10.18
+
+### Docker image support during flow deployment
+We have introduced enhanced support for Docker-based infrastructures when deploying flows through the interactive `prefect deploy` experience. Users can now easily custom-build or auto-build Docker images and push them to remote registries if they so choose.
```
Comment (znicholasbrown) [inline RELEASE-NOTES.md:6]:
```suggestion
We enhanced support for Docker-based infrastructures when deploying flows through the interactive `prefect deploy` experience. Users can now easily custom-build or auto-build Docker images and push them to remote registries if they so choose.
```

File: RELEASE-NOTES.md [hunk 20]
```diff
@@ -1,5 +1,56 @@
...
+
+The CLI automatically detects if a work pool supports Docker images (e.g., docker, ecs, cloud-run) during `prefect deploy`, guiding the user through the experience of building and pushing a Docker image.
```
Comment (znicholasbrown) [inline RELEASE-NOTES.md:8]:
```suggestion
The CLI automatically detects if a work pool supports Docker images (e.g., docker, ecs, cloud-run) during `prefect deploy` and will now guide the user through the experience of building and pushing a Docker image if support is detected.
```

File: RELEASE-NOTES.md [hunk 21]
```diff
@@ -1,5 +1,56 @@
...
+This enhancement to managing deployments will greatly simplify the process of creating the necessary `build` and `push` steps for deployments.
+
+Not only that, we will also create a `pull` step for you when you choose to build a Docker image through `prefect deploy`. Whether you have your own Dockerfile or you want to use the auto-build feature in `build_docker_image`, we will create a `pull` step for you to help you set the correct path to your flow code.
+
+See the following pull request for implementation details:
+- https://github.com/PrefectHQ/prefect/pull/10022
+- https://github.com/PrefectHQ/prefect/pull/10090
+
+### Event-driven deployments with triggers
+You can now add event-based triggers to your deployments in Prefect Cloud, allowing you to add triggers into your `prefect.yaml` file or from the Prefect UI deployment page. Deployment triggers use automations under the hood, and any automations that trigger a deployment will be linked to from the deployments page.
```
Comment (znicholasbrown) [inline RELEASE-NOTES.md:19]:
```suggestion
You can now easily incorporate event-based triggers into your Prefect Cloud deployments - simply add triggers to your `prefect.yaml` file or directly from the Prefect UI deployment page. Deployment triggers utilize automations - any automation that runs flows from a given deployment will be reflected on that deployment page.
```
I cleaned this up a little for clarity/brevity, feel free to reject this

**Review by znicholasbrown — APPROVED** [review]
A few trivial comments from me, mostly for brevity, LGTM

**Pull request merged**